### ISSUE TITLE
example cgi script in LAMP/LAPP appliances is not working

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#722]
The example cgi script included in the LAMP and LAPP appliances is not currently working. It appears that there was a significant Apache 2.4 config change that messed up our default config.

Thanks to Alex for reporting on the [forums](https://www.turnkeylinux.org/forum/support/20100208/turnkey-lamp-and-cgi#comment-26727)

**Comment by JedMeister** [comment]
See http://askubuntu.com/a/547400/253180

**Comment by JedMeister** [comment]
Here's the specifics on working around it:
````
OLD=usr/lib/cgi-bin
NEW=var/www/cgi-bin
FILE=/etc/apache2/conf-available/serve-cgi-bin.conf
sed -i "s|$OLD|$NEW|g" $FILE
````

Plus we need to enable the cgi module:
````
a2enmod cgi
````

And finally restart Apache:
````
service apache 2 restart
````

**Comment by whampt** [comment]
Not working

**Comment by whampt** [comment]
[1]  
https://www.turnkeylinux.org/forum/support/20100208/turnkey-lamp-and-cgi#comment-26727
[2] https://github.com/turnkeylinux/tracker/issues/722
[3] https://github.com/turnkeylinux/tracker/issues/722#issuecomment-257510083

Done all. Can work with as my mac shows the page correctly, but not my linux box. This is only a test server, so its kind of ok. Would really like to know what I need to do here to get apache to serve perl as cgi. Am very new to apache, and am trying to learn WAY to many things at once and I really want this to just work. thanks

server-cgi-bin.conf

<IfModule mod_alias.c>
        <IfModule mod_cgi.c>
                Define ENABLE_USR_LIB_CGI_BIN
        </IfModule>

        <IfModule mod_cgid.c>
                Define ENABLE_USR_LIB_CGI_BIN
        </IfModule>

        <IfDefine ENABLE_USR_LIB_CGI_BIN>
                ScriptAlias /cgi-bin/ /var/www/cgi-bin/
                <Directory "/var/www/cgi-bin">
                        AllowOverride None
                        Options +ExecCGI -MultiViews +SymLinksIfOwnerMatch
                        Require all granted
                </Directory>
        </IfDefine>
</IfModule>

# vim: syntax=apache ts=4 sw=4 sts=4 sr noet


000-default.conf

ServerName localhost

<VirtualHost *:80>
        ServerAdmin webmaster@localhost
        DocumentRoot /var/www/
</VirtualHost>

<VirtualHost *:443>
        SSLEngine on
        ServerAdmin webmaster@localhost
        DocumentRoot /var/www/

ScriptAlias /cgi-bin/ /var/www/cgi-bin/
<Directory "/var/www/cgi-bin">
        AllowOverride None
        Options +ExecCGI -Multiviews +SymLinksIfOwnerMatch 
        Require all granted    

        AddHandler cgi-script cgi
</Directory>
ErrorLog ${APACHE_LOG_DIR}/error.log
        CustomLog ${APACHE_LOG_DIR}/access.log combined
</VirtualHost>

**Comment by JedMeister** [comment]
When I posted my fix instructions above, I had tested this on TKL LAMP v14.1 and confirmed that it worked. So I'm not sure why it's not working for you?! Unfortunately I don't have a running server handy, but I'll try to double check 

I did noticed that I had set the paths of `OLD` and `NEW` as relative (to root - `/`) rather than absolute. I updated my previous post to use absolute paths, then realised that I had done that on purpose... Doh! So I changed it back...
